### PR TITLE
Attention dropout (p=0.1) on Transolver — cross-dataset regularization

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

**Attention dropout as a simple regularization lever across all CFD benchmarks.**

The Transolver attention mechanism has no dropout currently. Adding attention dropout (applied to attention weights after softmax, before the value aggregation) is a classic regularization technique that:

1. Forces each slice-attention head to not over-rely on any single position
2. Reduces co-adaptation between attention heads
3. Provides stochastic regularization that can help generalization to OOD conditions

This is an architectural regularization change that should be dataset-agnostic and tests whether the current models are overfit in attention space.

## Implementation

In the Transolver attention module, add `nn.Dropout` after the softmax:

```python
# In Physics_Attention (or equivalent) __init__:
self.attn_drop = nn.Dropout(p=0.1)

# In forward, after softmax:
attn_weights = torch.softmax(scores / math.sqrt(self.head_dim), dim=-1)
attn_weights = self.attn_drop(attn_weights)   # <-- add this line
out = torch.einsum('bhij,bhjd->bhid', attn_weights, v)
```

Apply this to all attention blocks in the model (slice attention and any cross-attention).

## Runs (3 datasets)

Run these 3 jobs sequentially or in parallel across the 8 GPUs. Use `--wandb_group attn-dropout-cross-dataset` on all runs.

### Run 1 — TandemFoil

```bash
python train.py \
  --dataset tandemfoil \
  --optimizer lion \
  --lr 1.25e-4 \
  --cosine-t-max 10 \
  --grad-clip 0.5 \
  --weight-decay 1e-2 \
  --model-slices 64 \
  --model-layers 3 \
  --model-hidden-dim 192 \
  --model-heads 3 \
  --enable-fourier \
  --enable-te-coord-frame \
  --enable-cp-panel \
  --enable-cp-panel-tandem-only \
  --asinh-pressure \
  --residual-prediction \
  --enable-pressure-prior-addition \
  --epochs 999 \
  --ema-decay 0.999 \
  --wandb_group attn-dropout-cross-dataset
```

### Run 2 — AirfRANS

```bash
python train.py \
  --dataset airfrans \
  --airfrans-task full \
  --optimizer adamw \
  --lr 6e-4 \
  --cosine-t-max 50 \
  --grad-clip 1.0 \
  --weight-decay 1e-2 \
  --no-use-ema \
  --enable-fourier \
  --model-layers 2 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --epochs 999 \
  --wandb_group attn-dropout-cross-dataset
```

### Run 3 — DrivAerML

```bash
SENPAI_MAX_EPOCHS=9999 python train.py \
  --dataset drivaerml \
  --optimizer adamw \
  --lr 5e-4 \
  --cosine-t-max 30 \
  --no-use-ema \
  --enable-fourier \
  --model-layers 4 \
  --model-hidden-dim 512 \
  --model-heads 8 \
  --epochs 999 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 \
  --max-eval-batches 200 \
  --wandb_group attn-dropout-cross-dataset
```

## Current Baselines

| Dataset | Metric | Current Best | Best PR |
|---------|--------|-------------|---------|
| TandemFoil | val_primary/surface_pressure_mae | 22.537 | #2924 |
| AirfRANS | val_primary/surface_mse | 0.000482 | #2951 |
| DrivAerML | val_primary/surface_rel_l2_pct | 3.997% | #2898 |

## Expected outcome

Attention dropout should act as an orthogonal regularizer — if the models have any overfitting in attention space, this will show up as improvements across all three benchmarks. TandemFoil is most likely to benefit given its smaller dataset size and more complex OOD structure.

## Reporting

Please report W&B run IDs and the best val_primary metric for each dataset in a PR comment. Include the final epoch for each run.

### CODE REVIEW

**Comment by morganmcg1** [comment]
Closing to narrow the queue for the final ICML benchmark sprint. Current priority is benchmark-specific optimization and benchmark-faithful reporting for DrivAerML, TandemFoil paper benchmark, AirfRANS full Surf/Vol, and the Tandem parity anchor. If this idea still matters later, we can reopen it in a benchmark-specific form.
